src/app/calculator/system-and-equipment-curve/system-and-equipment-curve-graph/system-and-equipment-curve-graph.component.ts

```diff
@@ -211,41 +188,69 @@ export class SystemAndEquipmentCurveGraphComponent implements OnInit {
 
   // Every update is newPlot to force resize
   initRenderChart(isResize = false) {
-    Plotly.purge(this.currentSystemChartId);
-    Plotly.purge(this.currentPowerChartId);
     this.setDisplayDataOptions();
+    this.dataPointTraces = new Array();
+    this.selectedDataPoints = new Array();
     this.initChartSetup(isResize);
     this.drawTraceData();
+    this.newPlot();
+    this.save();
+  }
 
+  newPlot() {
     let chartLayout = JSON.parse(JSON.stringify(this.curveEquipmentChart.layout));
-    Plotly.newPlot(this.currentSystemChartId, this.curveEquipmentChart.data, chartLayout, this.curveEquipmentChart.config)
-      .then(chart => {
-        chart.on('plotly_click', (graphData) => {
-          this.createDataPoint(graphData);
-        });
-        chart.on('plotly_hover', hoverData => {
-              this.displayHoverData(hoverData);
-        });
-        chart.on('plotly_unhover', unhoverData => {
-            this.removeHoverData(this.currentPowerChartId);
-        });
+    let traceData: Array<TraceData> = new Array();
+    this.curveEquipmentChart.data.forEach((trace, i) => {
+      traceData.push(trace);
     });
+    this.dataPointTraces.forEach(trace => {
+      traceData.push(trace);
+    })
+    if (this.expanded && this.expandedSystemChartDiv) {
+      this.plotlyService.newPlot(this.expandedSystemChartDiv.nativeElement, traceData, chartLayout, this.curveEquipmentChart.config)
+        .then(chart => {
+          chart.on('plotly_click', (graphData) => {
+            this.createDataPoint(graphData);
```

Comment:
No hover data showing in in table when expanded. Looks like the event listened and method just need to be added for the expanded block

Reply:
fixed